src/server.js

```diff
@@ -48,9 +49,41 @@ module.exports = function() {
 			maxAge: 86400 * 1000,
 		}));
 
+	app.get("/uploads/:name/:slug*?", (req, res) => {
+		const name = req.params.name;
+		const folder = req.params.name.substring(0, 2);
+		const uploadPath = Helper.getFileUploadPath();
+		const filePath = path.join(uploadPath, folder, name);
+		const nameRegex = /^[0-9a-f]{16}$/;
+
+		if (!nameRegex.test(name)) {
+			return res.status(404).send("Not found");
+		}
+
+		const fileType = Uploader.getFileType(filePath);
+		let mimeType = "application/octet-stream";
+		let contentDisposition = "attachment";
+
+		// doesn't exist
+		if (fileType === undefined) {
+			return res.status(404).send("Not found");
+		}
+
+		// If whitelisted
+		if (fileType && fileType.mime && Uploader.isValidType(fileType.mime)) {
+			mimeType = fileType.mime;
```

Comment:
I think `mimeType` should be set regardless. And `Uploader.isValidType` will control `contentDisposition`.

Are we going to look into using `is-utf8` as we discussed?